### PR TITLE
Pilot-agent: remove outdated envs

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
remove  `SKIP_PARSE_TOKEN`, `PLUGINS` env.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
part of ##28047

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -126,10 +126,6 @@ var (
 	enableGatewaySDSEnv = env.RegisterBoolVar("ENABLE_INGRESS_GATEWAY_SDS", false,
 		"Enable provisioning gateway secrets. Requires Secret read permission").Get()
 
-	// TODO: This is already present in ProxyConfig !!!
-	trustDomainEnv = env.RegisterStringVar("TRUST_DOMAIN", "",
```
Comment (irisdingbj) [inline pilot/cmd/pilot-agent/main.go:130]: shall we remove `trust-domain` in manifest like https://github.com/istio/istio/blob/master/manifests/charts/istio-control/istio-discovery/files/injection-template.yaml#L165 as well for this clean up?

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/main.go:130]: thanks for pointing it

File: pilot/cmd/pilot-agent/config.go [hunk 2]
```diff
@@ -33,23 +33,25 @@ import (
...
+			return meshconfig.ProxyConfig{}, trustDomain, fmt.Errorf("failed to read mesh config file %v: %v", meshConfigFile, err)
 		}
 		fileMeshContents = string(contents)
 	}
 	meshConfig, err := getMeshConfig(fileMeshContents, annotations[annotation.ProxyConfig.Name])
 	if err != nil {
-		return meshconfig.ProxyConfig{}, err
+		return meshconfig.ProxyConfig{}, trustDomain, err
 	}
+	trustDomain = meshConfig.TrustDomain
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/config.go:54]: This is not safe. Only gateway has mesh config mounted. Proxies can only rely on ProxyConfig

File: pilot/cmd/pilot-agent/main.go [hunk 3]
```diff
@@ -118,18 +117,10 @@ var (
...
-	// TODO: this is a horribly named env, it's really TOKEN_EXCHANGE_PLUGINS - but to avoid breaking
-	// it's left unchanged. It may not be needed because we autodetect.
-	pluginNamesEnv = env.RegisterStringVar("PLUGINS", "", "Token exchange plugins").Get()
-
 	// This is also disabled by presence of the SDS socket directory
 	enableGatewaySDSEnv = env.RegisterBoolVar("ENABLE_INGRESS_GATEWAY_SDS", false,
 		"Enable provisioning gateway secrets. Requires Secret read permission").Get()
 
-	// TODO: This is already present in ProxyConfig !!!
-	trustDomainEnv = env.RegisterStringVar("TRUST_DOMAIN", "",
```
Comment (howardjohn) [inline pilot/cmd/pilot-agent/main.go:130]: Please do not remove this, see comment about mesh config above.

**Review by howardjohn — CHANGES_REQUESTED** [review]
LGTM if you revert the TRUST_DOMAIN removal. We can remove the other stuff and `--trust-domain` though

File: pilot/cmd/pilot-agent/config.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/config.go:54]: Thanks for the news. But just want to know why the proxy can have different trust-domain.

**Review by howardjohn — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
rebased @irisdingbj could you approve

**Review by irisdingbj — APPROVED** [review]

**Pull request merged**